compiler/src/iree/compiler/GlobalOptimization/SetEncoding.cpp

```diff
@@ -214,6 +191,24 @@ static Value unsetEncodingAndExtractSlice(OpBuilder &builder, Location loc,
                                                 sizes, strides);
 }
 
+/// Given a LinalgOp and one of its OpOperands, return the element type,
+/// inferring unsignedness from the body of the LinalgOp
+static Type getContractionInputTypeWithSignedness(OpBuilder &builder,
+                                                  linalg::LinalgOp linalgOp,
+                                                  OpOperand *operand) {
+  assert(linalg::isaContractionOpInterface(linalgOp));
+  assert(operand->getOwner() == linalgOp.getOperation());
+  auto elemType = getElementTypeOrSelf(operand->get().getType());
+  // Infer if unsigned from body ops
+  Value blockArg = linalgOp.getMatchingBlockArgument(operand);
+  for (auto bodyCastOp : blockArg.getParentBlock()->getOps<arith::ExtUIOp>()) {
+    if (bodyCastOp->getOperand(0) == blockArg) {
+      return builder.getIntegerType(elemType.getIntOrFloatBitWidth(), false);
```

Comment:
nit: add argument comment

```suggestion
      return builder.getIntegerType(elemType.getIntOrFloatBitWidth(), /*isSigned=*/false);
```